File: ios/SampleAlert.m
//
//  SampleAlert.m
//  ExazeEvaluation
//
//
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(SampleAlert, NSObject)
RCT_EXTERN_METHOD(alertShow)
RCT_EXTERN_METHOD(alertClose)
RCT_EXTERN_METHOD(getStatus: (RCTResponseSenderBlock)callback)
@end

File: ios/SampleAlert.swift
//
//  SampleAlert.swift
//  ExazeEvaluation
//

import Foundation
@objc(SampleAlert)
class SampleAlert: NSObject {
  @objc
  static var isAlertShown = false
  
  @objc
  func alertShow() {
    SampleAlert.isAlertShown = true
    print("Alert is now shown")
  }
  
  @objc
  func alertClose() {
    SampleAlert.isAlertShown = false
    print("Alert is now closed")
  }
  
  @objc
  func getStatus(_ callback: RCTResponseSenderBlock) {
    callback([NSNull(), SampleAlert.isAlertShown])
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
}